ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java

```diff
@@ -526,7 +581,7 @@ public static String canonicalType(String type) {
                     try {
                         String typeName = canonicalType(types.unpackString());
 
-                        if (typeName.length() == 0)
+                        if (typeName == null || typeName.length() == 0)
```

Comment:
`types.unpackString()` can never be null unless someone packed a zero length string in the object store, why would someone do that? If you think we should protect against this eventuality then something is corrupt or boken so you may as well throw an ObjectStoreException and handle it in the catch block (and change the log message from trace to warn).

Reply:
I see. This change came was rather from my lack of understanding. I took it from the fact that method `canonicalType` may return the `null` and thus it seemed to me that the method considers the null type string as a possible option.
I will remove my adjustment as it's not necessary.